spring-rabbit/src/test/java/org/springframework/amqp/rabbit/annotation/EnableRabbitReturnTypesTests.java

```diff
@@ -117,12 +121,35 @@ public CachingConnectionFactory cf() {
 			return new CachingConnectionFactory(RabbitAvailableCondition.getBrokerRunning().getConnectionFactory());
 		}
 
+		@Bean
+		public RabbitAdmin admin(CachingConnectionFactory cf) {
+			return new RabbitAdmin(cf);
+		}
+
 		@Bean
 		public Jackson2JsonMessageConverter converter() {
 			return new Jackson2JsonMessageConverter();
 		}
 
-		@RabbitListener(queues = "EnableRabbitReturnTypesTests.1")
+		@Bean
+		public SimpleAsyncTaskExecutor exec() {
+			return new SimpleAsyncTaskExecutor();
+		}
+
+		@Bean
+		public ReplyPostProcessor rpp() {
+			return (in, out) -> out;
+		}
+
+		@Bean
+		public RabbitListenerErrorHandler rleh() {
+			return (amqpMessage, message, exception) -> null;
+		}
+
+		@RabbitListener(queues = "EnableRabbitReturnTypesTests.1", admin = "#{admin}",
+				containerFactory = "#{@rabbitListenerContainerFactory}",
+				executor = "#{@exec}", replyPostProcessor = "#{@rpp}", messageConverter = "#{converter}",
+				errorHandler = "#{@rleh}")
```

Comment:
Why difference in these expression? Why some of them with the `@`, but other not?
Are they evaluated to a different result? Like some to bean references, but other to their bean name?
Thanks

Reply:
Interesting; looks like the `@` is optional here. I will investigate.

Reply:
Yes, the `@` is optional here (resolved by the `BeanExpressionContextAccessor` if not there and `BeanFactoryResolver` if present).